Description: call inet_pton on the real addr
 The user may pass in a "addr/mask" value.  inet_pton only accepts
 addr.  We make a copy of the address with the '/' set to '\0', but
 then call inet_pton on the original, not sanitized address.  Fix
 that.
Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/lxc/+bug/1215391
Forwarded: yes

Index: lxc-0.9.0/src/lxc/confile.c
===================================================================
--- lxc-0.9.0.orig/src/lxc/confile.c	2013-08-23 12:29:32.219873001 -0500
+++ lxc-0.9.0/src/lxc/confile.c	2013-08-23 12:38:50.269873000 -0500
@@ -745,8 +745,8 @@
 		inet6dev->prefix = atoi(netmask);
 	}
 
-	if (!inet_pton(AF_INET6, value, &inet6dev->addr)) {
-		SYSERROR("invalid ipv6 address: %s", value);
+	if (!inet_pton(AF_INET6, valdup, &inet6dev->addr)) {
+		SYSERROR("invalid ipv6 address: %s", valdup);
 		free(valdup);
 		return -1;
 	}
